pkg/cmd/cli/cmd/process.go

```diff
@@ -136,6 +112,14 @@ func RunProcess(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []
 		}
 	}
 
+	env, duplicates, errs := cmdutil.ParseEnvironmentArgumentsFiles(valueArgs)
+	if len(errs) > 0 {
+		return kerrors.NewAggregate(errs)
+	}
+	if len(duplicates) > 0 {
+		return kcmdutil.UsageError(cmd, fmt.Sprintf("The following values were provided more than once: %s", strings.Join(duplicates, ", ")))
+	}
+
```

Comment:
Done, I've also moved the block that checks for incompatible options to the very beginning so that the error behavior doesn't change too much.